api/queries_repo.go

```diff
@@ -782,35 +783,54 @@ func (m *RepoMetadataResult) projectV2TitleToID(projectTitle string) (string, bo
 	return "", false
 }
 
-func ProjectsToPaths(projects []RepoProject, projectsV2 []ProjectV2, names []string) ([]string, error) {
-	var paths []string
-	for _, projectName := range names {
-		found := false
-		for _, p := range projects {
-			if strings.EqualFold(projectName, p.Name) {
-				// format of ResourcePath: /OWNER/REPO/projects/PROJECT_NUMBER or /orgs/ORG/projects/PROJECT_NUMBER or /users/USER/projects/PROJECT_NUBER
-				// required format of path: OWNER/REPO/PROJECT_NUMBER or ORG/PROJECT_NUMBER or USER/PROJECT_NUMBER
-				var path string
-				pathParts := strings.Split(p.ResourcePath, "/")
-				if pathParts[1] == "orgs" || pathParts[1] == "users" {
-					path = fmt.Sprintf("%s/%s", pathParts[2], pathParts[4])
-				} else {
-					path = fmt.Sprintf("%s/%s/%s", pathParts[1], pathParts[2], pathParts[4])
+func ProjectNamesToPaths(client *Client, repo ghrepo.Interface, projectNames []string, projectsV1Support gh.ProjectsV1Support) ([]string, error) {
```

Comment:
This should be functionally the same as the previous `ProjectNamesToPaths` but avoids requesting projectv1 if not supported.

Reply:
Since there's already a `name` field on projects, I'd rename this to `ProjectTitlesToPaths` to match the related field name (`ProjectTitles`) to avoid confusion. If you renamed this, then do the same for the `projectNames` parameter.